maat-court-data-api/src/main/java/gov/uk/courtdata/prosecutionconcluded/service/ProsecutionConcludedService.java

```diff
@@ -1,16 +1,87 @@
+import gov.uk.courtdata.repository.WQHearingRepository;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
 
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
 @Service
 @RequiredArgsConstructor
 @Slf4j
 public class ProsecutionConcludedService {
 
+    private final CalculateCrownCourtOutcome calculateCrownCourtOutcome;
+
+    private final WQHearingRepository wqHearingRepository;
+
+    private final OUCodeValidator ouCodeValidator;
+    private final CaseTypeValidator caseTypeValidator;
+
+    private final ProsecutionConcludedDAO prosecutionConcludedDAO;
+
     public void execute(final ProsecutionConcluded prosecutionConcluded) {
 
+        WQHearingEntity wqHearingEntity = wqHearingRepository.getById(prosecutionConcluded.getProsecutionCaseId().toString());
+
+        if (wqHearingEntity.getMaatId()==null)
+            throw new MAATCourtDataException("Hearing not found for this hearingId " + prosecutionConcluded.getProsecutionCaseId());
+
+        if (JurisdictionType.CROWN.name().equalsIgnoreCase(wqHearingEntity.getWqJurisdictionType())) {
+
+            String calculatedOutcome = calculateCrownCourtOutcome.calculate(prosecutionConcluded);
+            log.info("calculated outcome is {} for this maat-id {}", calculatedOutcome, prosecutionConcluded.getMaatId());
+
+            //Todo: create or update the existing classes to process this validation
+            //ouCodeValidator.validate(wqHearingEntity.getOuCourtLocation());
+            //caseTypeValidator.validate(maatId, calculatedOutcome);
+
+            ConcludedDTO concludedDTO = ConcludedDTO.
+                    builder()
+                    .prosecutionConcluded(prosecutionConcluded)
+                    .calculatedOutcome(calculatedOutcome)
+                    .ouCourtLocation(wqHearingEntity.getOuCourtLocation())
+                    .wqJurisdictionType(wqHearingEntity.getWqJurisdictionType())
+                    .appealType("AR")
+                    .caseEndDate(getMostRecentCaseEndDate(prosecutionConcluded.getOffenceSummaryList()))
+                    .caseUrn(wqHearingEntity.getCaseUrn())
+                    .hearingResultCode(buildResultCodeList(wqHearingEntity))
+                    .build();
+
+            prosecutionConcludedDAO.execute(concludedDTO);
+        }
+    }
+
+    private String getMostRecentCaseEndDate(List<OffenceSummary> offenceSummaryList) {
+
+        if(offenceSummaryList.isEmpty())
+            return null;
+
+        return offenceSummaryList.stream()
+                .map(offenceSummary -> LocalDate.parse(offenceSummary.getProceedingsConcludedChangedDate()))
+                .distinct()
+                .collect(Collectors.toList())
+                .stream().sorted()
+                .collect(Collectors.toList()).get(0).toString();
```

Comment:
I think this isn't picking up the most recent one. It will pick up the least recent one as the sort is ASC. Can we also verify the format of the date from CDA & check the parsing here?